# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/man-pages/sharedfiles.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -Naur man-pages-1.70.orig/Makefile man-pages-1.70/Makefile
--- man-pages-1.70.orig/Makefile	2004-06-24 18:49:38.000000000 +0100
+++ man-pages-1.70/Makefile	2004-12-20 15:59:05.000000000 +0000
@@ -16,11 +16,12 @@
 
 screen:
 	-mkdir not_installed
+	find /var/adm/flists ! -name "man-pages*" | \
+	xargs grep -hE "/man/man([1-9])/.*\.\1$$" | \
+	awk '{print $$2}' > not_installed/.MANCACHE; \
 	for i in man?/*; do \
-		if [ $(MANDIR)/"$$i" -nt "$$i" ]; then \
-			cmp -s $(MANDIR)/"$$i" "$$i" > /dev/null 2>&1; \
-			if [ "$$?" != 0 ]; then mv "$$i" not_installed; fi; \
-		fi; \
+		grep -Fq "/$$i" not_installed/.MANCACHE; \
+		if [ "$$?" == 0 ]; then mv "$$i" not_installed; fi; \
 	done
 
 remove:
